### ISSUE TITLE
constinferred broken on Julia 1.6

### ISSUE THREAD

**Issue by mhauru** [Jutho/TestExtras.jl#1]
A recent commit https://github.com/JuliaLang/julia/commit/5a56ecdd767709c844f65d96be4d35c35698cbc3 to Julia changed the call signature of `typesubtract`: It now requires a third argument, `MAX_UNION_SPLITTING::Int`. I don't understand why it's not an optional argument with a sensible default, but it's not. This breaks `constinferred`.

**Comment by Jutho** [comment]
It seems this was solved for `Test.@inferred` by changing the dependence on `Core.Compiler.typesubtract` to a function call to a new function `Base.typesplit`. I will fix TestExtras.jl in the same way. I should anyway better understand what exactly this is doing anyway, as I've now just copy-pasted this from `@inferred`.

**Comment by mhauru** [comment]
The source of `typesplit` comes with the following warning:
```
# WARNING: this is wrong for some objects for which subtyping is broken
#          (Core.Compiler.isnotbrokensubtype), use only simple types for `b`
```
I don't know if that's of concern. I guess not if that's what `@inferred` is doing.

**Comment by Jutho** [comment]
This is now fixed; and version 0.1 has been registered, but this will of course take 3 days before being in effect. I will then switch the tests of TensorKit.jl to depend on TestExtras.jl.
